src/python/WMCore/Storage/StageInMgr.py

```diff
@@ -204,91 +213,51 @@ def __call__(self, **fileToStage):
         option - the option values to be passed to that command (None is allowed)
         lfn-prefix - the LFN prefix to generate the PFN
         phedex-node - the Name of the PNN to which the file is being xferred
-
         """
         localPfn = os.path.join(os.getcwd(), os.path.basename(lfn))
-
-        if override:
-            pnn = override['phedex-node']
-            command = override['command']
-            options = override['option']
-            pfn = "%s%s" % (override['lfn-prefix'], lfn)
+        if self.override:
+            pnn = self.overrideConf['phedex-node']
+            command = self.overrideConf['command']
+            options = self.overrideConf['option']
+            pfn = "%s%s" % (self.overrideConf['lfn-prefix'], lfn)
             protocol = command
         else:
-            pnn = self.siteCfg.localStageOut['phedex-node']
-            command = self.siteCfg.localStageOut['command']
-            options = self.siteCfg.localStageOut.get('option', None)
-            pfn = self.searchTFC(lfn)
-            protocol = self.tfc.preferredProtocol
+            pnn = stageOut_rfc[0]['phedex-node']
+            command = stageOut_rfc[0]['command']
+            options = stageOut_rfc[0]['option']
+            pfn = searchRFC(stageOut_rfc[1],lfn)
+            protocol = stageOut_rfc[1].preferredProtocol
 
         if pfn == None:
             msg = "Unable to match lfn to pfn: \n  %s" % lfn
-            raise StageOutFailure(msg, LFN = lfn, TFC = str(self.tfc))
-
+            raise StageOutFailure(msg, LFN=lfn, StageOut=stageOutStr(stageOut_rfc[0]))
         try:
-            impl = retrieveStageOutImpl(command, stagein=True)
+            impl = retrieveStageOutImpl(command,stagein=True)
         except Exception as ex:
             msg = "Unable to retrieve impl for local stage in:\n"
-            msg += "Error retrieving StageOutImpl for command named: %s\n" % (
-                command,)
-            raise StageOutFailure(msg, Command = command,
-                                  LFN = lfn, ExceptionDetail = str(ex))
+            msg += "Error retrieving StageOutImpl for command named: %s\n" % (command,)
+            raise StageOutFailure(msg, Command=command,
+                                  LFN=lfn, ExceptionDetail=str(ex))
         impl.numRetries = self.numberOfRetries
         impl.retryPause = self.retryPauseTime
 
         try:
-            impl(protocol, pfn, localPfn, options)
+            if not self.bypassImpl:
+                impl(protocol, pfn, localPfn, options)
         except Exception as ex:
-            msg = "Failure for local stage in:\n"
+            msg = "Failure for stage in:\n"
             msg += str(ex)
-            raise StageOutFailure(msg, Command = command, Protocol = protocol,
-                                  LFN = lfn, InputPFN = localPfn,
-                                  TargetPFN = pfn)
-
-        return localPfn
-
-
-    def searchTFC(self, lfn):
```

Comment:
I am just connecting the dots here. This method has been replaced by `StageOutMgr.searchRFC`, right?

Reply:
Yes